### PR TITLE
screen setup

### PR DESCRIPTION
**Pull request by jspauldo98**
added an abstract class Screen.java which will be each screen's superclass

added ScreenManager.java class to control different screens

added a blank MenuScreen.java class extending Screen.java

revised Main.java so that it implements ApplicationListener and set to create, resize, and render MenuScreen.java

---
This change is [Reviewable](https://reviewable.io/reviews/boomboxinc/salt/27)

### CODE REVIEW

File: core/src/com/spaulding/salt/Main.java [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
+import com.spaulding.salt.Screen.MenuScreen;
+import com.spaulding.salt.Screen.ScreenManager;
 
-public class Main extends ApplicationAdapter {
-	SpriteBatch batch;
-	Texture img;
+public class Main implements ApplicationListener {
+	Stage stage;
 	
 	@Override
```
Comment (ConnorGBrewster) [inline core/src/com/spaulding/salt/Main.java]: I don't think these overrides need to be here, since this isn't implementing `Screen` nor should it.

**Comment by ConnorGBrewster** [comment]
@homu-salt try

**Comment by homu-salt** [comment]
:hourglass: Trying commit d7f8faf with merge d415fc8...

**Comment by homu-salt** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/boomboxInc/Salt/builds/129869779)

File: core/src/com/spaulding/salt/Main.java (hunk 1 above)
Comment (ConnorGBrewster) [inline core/src/com/spaulding/salt/Main.java]: https://github.com/libgdx/libgdx/wiki/The-life-cycle

**Comment by ConnorGBrewster** [comment]
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[core/src/com/spaulding/salt/Main.java, line 13 \[r1\]](https://reviewable.io:443/reviews/boomboxinc/salt/27#-KHbAl5sTViS-ftSIq1A-r1-13:-KHbjmOxYhJbci6dPBLc:663549890) ([raw file](https://github.com/boomboxinc/salt/blob/d7f8faf2b6f3a30cba4b995fa7316739a4d121b7/core/src/com/spaulding/salt/Main.java#L13)):*
<details><summary><i>Previously, ConnorGBrewster (Connor Brewster) wrote…</i></summary>
> https://github.com/libgdx/libgdx/wiki/The-life-cycle 
> 
> We only need @Override if extending `ApplicationAdapter`, we do not need overrides when implementing `ApplicationListener`. The `ApplicationAdapter` extends `ApplicationListener` and has some default implementation.</details>
Since we have the methods here triggering the methods on the current screen, I think it is ok to stick with `ApplicationListener`. So let's just get rid of the `@Override`s. After the ScreenManager change, we can store a class level ScreenManager here and create a new instance on `create()`.

---

*[core/src/com/spaulding/salt/Screen/MenuScreen.java, line 10 \[r1\]](https://reviewable.io:443/reviews/boomboxinc/salt/27#-KHbjcckqsf-5tz3Yy4L:-KHbjcckqsf-5tz3Yy4M:-1721315216) ([raw file](https://github.com/boomboxinc/salt/blob/d7f8faf2b6f3a30cba4b995fa7316739a4d121b7/core/src/com/spaulding/salt/Screen/MenuScreen.java#L10)):*
> ```Java
> public class MenuScreen extends Screen {
> 
>     @Override
> ```

I believe these don't need to be `@Override` because Screen never gave a default implementation. This would be the first time they are implemented.

---

*[core/src/com/spaulding/salt/Screen/ScreenManager.java, line 8 \[r1\]](https://reviewable.io:443/reviews/boomboxinc/salt/27#-KHbkKlgSCLF1rBL7wrj:-KHbkKlgSCLF1rBL7wrk:-1639731858) ([raw file](https://github.com/boomboxinc/salt/blob/d7f8faf2b6f3a30cba4b995fa7316739a4d121b7/core/src/com/spaulding/salt/Screen/ScreenManager.java#L8)):*
> ```Java
> public class ScreenManager {
> 
>     private static Screen current_screen;
> ```

If we could avoid making this `ScreenManager` a singleton type class, that would be awesome. So I would make the get/set current scene not be static and the scene class level variable not be static, and then add a constructor. There are pros/cons on using singletons, personally if I can avoid them, I prefer that as they can be tough to work with/reason about when a project gets larger.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/boomboxinc/salt/27)*

**Comment by ConnorGBrewster** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/boomboxinc/salt/27)*

**Comment by ConnorGBrewster** [comment]
@homu-salt r+

**Comment by homu-salt** [comment]
:pushpin: Commit e03c946 has been approved by `ConnorGBrewster`

**Comment by homu-salt** [comment]
:hourglass: Testing commit e03c946 with merge 8e346a5...

**Comment by homu-salt** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/boomboxInc/Salt/builds/130094768)

**Pull request merged**